src/java.base/share/classes/java/text/ListFormat.java

```diff
@@ -0,0 +1,557 @@
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj instanceof ListFormat other) {
+            return locale.equals(other.locale) &&
+                Arrays.equals(patterns, other.patterns);
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int hashCode() {
+        return Objects.hash(locale, Arrays.hashCode(patterns));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String toString() {
+        return
+            """
+            ListFormat [locale: "%s", start: "%s", middle: "%s",  end: "%s", two: "%s", three: "%s"]
+            """.formatted(locale.getDisplayName(), patterns[START], patterns[MIDDLE], patterns[END], patterns[TWO], patterns[THREE]);
+    }
+
+    private MessageFormat generateMessageFormat(Object[] input) {
+        var len = input.length;
+        return switch (len) {
+            case 0 -> throw new IllegalArgumentException("There should at least be one input string");
+            case 1 -> new MessageFormat("{0}", locale);
+            case 2, 3 -> {
+                var pattern = patterns[len + 1];
+                if (pattern != null && !pattern.isEmpty()) {
+                    yield new MessageFormat(pattern, locale);
+                } else {
+                    yield new MessageFormat(createMessageFormatString(len), locale);
+                }
+            }
+            default -> new MessageFormat(createMessageFormatString(len), locale);
+        };
+    }
+
+    private String createMessageFormatString(int count) {
+        var sb = new StringBuilder(patterns[START]);
+        IntStream.range(2, count - 1).forEach(i -> sb.append(middleBetween).append("{").append(i).append("}"));
+        sb.append(patterns[END].replaceFirst("\\{0}", "").replaceFirst("\\{1}", "\\{" + (count - 1) + "\\}"));
+        return sb.toString();
+    }
+
+    @java.io.Serial
+    private void readObject(ObjectInputStream stream)
+            throws IOException, ClassNotFoundException {
+        stream.defaultReadObject();
+        init();
+    }
+
+    /**
+     * A list format type - {@code standard}, {@code or}, and {@code unit}.
+     * <p>
+     * {@code Type} is an enum which represents the type for formatting
+     * a list within a given {@code ListFormat} instance.
+     */
+    public enum Type {
+
+        /**
+         * The {@code STANDARD} list format style. This is the default
+         * type, which concatenates elements in "and" enumeration.
+         */
+        STANDARD,
+
+        /**
+         * The {@code OR} list format style. This style concatenates
+         * elements in "or" enumeration.
+         */
+        OR,
+
+        /**
+         * The {@code UNIT} list format style. This style concatenates
+         * elements, useful for enumerating units.
+         */
+        UNIT;
+
+        @Override
+        public String toString() {
+            return name().toLowerCase(Locale.ROOT);
```

Comment:
I think the enum should be a regular java enum (no overrides to change the case).

Reply:
OK, this was so just to match the CLDR types. Agree it should align with other enums.